### ISSUE TITLE
Cyrtic error in xCode when using MobileAdvert plugin

### ISSUE THREAD

**Issue by Madiosn** [Scirra/Construct-3-bugs#2909]
## Problem description
When trying to customize ads in construct 3 using the standard MobileAdvert module, a critical error occurs in Xcode.

## Attach a .c3p
https://yadi.sk/d/X9DneYz1cJyELA
The project is normal. just added a module.

This is a prepared Xcode file.
https://yadi.sk/d/BmMoPk227volvQ

## Steps to reproduce
It is necessary to open the New project.xcodeproj file in Xcode (the file with the .xcodeprof extension), because only from it you can save the application for sending to the server.

Next, we assemble the application.

Further the error which I will put out further gets out.

1. It is necessary to open the New project.xcodeproj file in Xcode
2. Press Build
3. 

## Observed result

Showing Recent Messages
Ld /Users/maxmadison/Library/Developer/Xcode/DerivedData/New_project-elcjgrhrpprbvmepaxgxzmqfulbd/Build/Products/Debug-iphoneos/New\ project.app/New\ project normal arm64
    cd "/Users/maxmadison/Downloads/New project.ios.project"
    export IPHONEOS_DEPLOYMENT_TARGET=12.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Xlinker -rpath -Xlinker /usr/lib/swift -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -L/Users/maxmadison/Library/Developer/Xcode/DerivedData/New_project-elcjgrhrpprbvmepaxgxzmqfulbd/Build/Products/Debug-iphoneos -F/Users/maxmadison/Library/Developer/Xcode/DerivedData/New_project-elcjgrhrpprbvmepaxgxzmqfulbd/Build/Products/Debug-iphoneos -F/Users/maxmadison/Downloads/New\ project.ios.project/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks -filelist /Users/maxmadison/Library/Developer/Xcode/DerivedData/New_project-elcjgrhrpprbvmepaxgxzmqfulbd/Build/Intermediates.noindex/New\ project.build/Debug-iphoneos/New\ project.build/Objects-normal/arm64/New\ project.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -miphoneos-version-min=12.0 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/maxmadison/Library/Developer/Xcode/DerivedData/New_project-elcjgrhrpprbvmepaxgxzmqfulbd/Build/Intermediates.noindex/New\ project.build/Debug-iphoneos/New\ project.build/Objects-normal/arm64/New\ project_lto.o -Xlinker -no_deduplicate -fobjc-arc -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos -Xlinker -add_ast_path -Xlinker /Users/maxmadison/Library/Developer/Xcode/DerivedData/New_project-elcjgrhrpprbvmepaxgxzmqfulbd/Build/Intermediates.noindex/New\ project.build/Debug-iphoneos/New\ project.build/Objects-normal/arm64/New_project.swiftmodule -ObjC -ObjC -framework AVFoundation -framework AudioToolbox -framework CFNetwork -framework CoreGraphics -framework CoreMedia -framework CoreMotion -framework CoreTelephony -framework CoreVideo -framework GLKit -framework GoogleMobileAds -framework MediaPlayer -framework MessageUI -framework MobileCoreServices -framework OpenGLES -framework QuartzCore -framework Security -framework StoreKit -framework SystemConfiguration -weak_framework AdSupport -weak_framework JavaScriptCore -weak_framework SafariServices -weak_framework WebKit /Users/maxmadison/Library/Developer/Xcode/DerivedData/New_project-elcjgrhrpprbvmepaxgxzmqfulbd/Build/Products/Debug-iphoneos/libCordova.a -weak_framework WebKit -lPods-New\ project -Xlinker -dependency_info -Xlinker /Users/maxmadison/Library/Developer/Xcode/DerivedData/New_project-elcjgrhrpprbvmepaxgxzmqfulbd/Build/Intermediates.noindex/New\ project.build/Debug-iphoneos/New\ project.build/Objects-normal/arm64/New\ project_dependency_info.dat -o /Users/maxmadison/Library/Developer/Xcode/DerivedData/New_project-elcjgrhrpprbvmepaxgxzmqfulbd/Build/Products/Debug-iphoneos/New\ project.app/New\ project

ld: library not found for -lPods-New project
clang: error: linker command failed with exit code 1 (use -v to see invocation)

## More details

**Affected browsers/platforms:** 
I use
Xcode latest version.
MacOs latest version

**First affected release:** 
r146 / r147/
## System details

<details><summary>View details</summary>

PASTE HERE
Screen https://yadi.sk/i/Tt4bdQ6XxvzwTA
</details>

**Comment by shortercode** [comment]
@Madiosn you should use the workspace file, NOT the project file. Does this resolve your issue?

**Comment by Madiosn** [comment]
This solves the problem locally. That is, if I use the .xcworkspace file, this problem does not appear. But there is another problem. Which does not allow porting the application to Appstore connect. Since the .xcworkspace extension cannot be correctly assembled.

**Comment by shortercode** [comment]
It's not clear what you mean by "[...]cannot be correctly assembled". I presume you mean it has a compilation error. 

There are a few unpredictable issues related to building with xcode. You should be able to fix these by adjusting some project settings in xcode.

So I can advise you further could you please post details related to your current issue?

**Comment by Madiosn** [comment]
I have been struggling with this for the third day and doing various tests trying to catch the source of the error.

I could not solve the problem but I see that it is caused by the Mobile Advertising plugin.

What did I do.

1. Create in Construct 3 any application without advertising module.
1.1. We export it and load it into the app connector - everything is fine!

2. add an ad module to the application
2.1 open it in xCode as usual using the Edless Agent.xcodeproj file
2.2 trying to build the application and see the error (which is listed above)

3. We are trying to do the same, but now we open the file with the extension Edless Agent.xcworkspace
3.1 Making the assembly. The error is ignored and everything is fine. Build ready to ship to Apple Connect
3.2 Downloading the link to Apple Connect and we get an error at the end of the download: App Store ERROR ITMS-90174: "App Storeline - Apps have been installed."
3.3. This error is understandable; it appears when the Legace Build System is not specified in the Build System parameter. We can set this setting only from the .xcodeproj file (And it is configured there) But, overlooking the lack of it from the .xcworkspace file, it ignores this. Thus, we can not unload the application.

**Comment by shortercode** [comment]
You have to use the workspace file for the cocoapods dependencies to be included in the build. It's a restriction for any project using the cocoapods dependency manager ( which is quite common ). You can swap which version of the build system your using by using the "Workspace settings" dialog in xcode.

**Comment by Madiosn** [comment]
Да конечно я использую Версия 10.2.1 (10E1001)

**Comment by shortercode** [comment]
I mean the "Build System" version

**Comment by Madiosn** [comment]
Screen - this?
https://yadi.sk/i/-3PZMmf25VSrug

**Comment by Madiosn** [comment]
I know I need to use the Legacy version. And I use it. But as I said above, if the project is open from .xcworkspace

This parameter is not active.

https://yadi.sk/i/16c0Vi3wJFiLgw

**Comment by Madiosn** [comment]
It all worked. Thanks a lot @shortercode horsecode, 3 sleepless nights, eyes have not seen the little things. I checked and changed Lagace more than once, but it can be seen somewhere that I was just so sure of it that I did not notice. I have done over 40 application builds. I don't know what I would do without you.

I also had to change the names of all the sprites to English. Now everything works. I do not know why before this error as nebylo.

As always, thank you so much for not abandoning your subscribers!) You are the best!
